lib/bech32/test/Codec/Binary/Bech32Spec.hs

```diff
@@ -254,17 +257,27 @@ spec = do
             isJust (Bech32.toBase256 ws) ==>
                 (Bech32.toBase32 <$> Bech32.toBase256 ws) === Just ws
 
-    describe "Roundtrip (charToWord5 . word5ToChar)" $ do
-        it "can perform roundtrip character set conversion (lower-case)" $
+    describe "Roundtrip (dataCharToWord . dataCharFromWord)" $ do
+        it "can perform roundtrip character set conversion" $
             property $ \w ->
-                Bech32.charToWord5 (toLower (Bech32.word5ToChar Arr.! w))
+                Bech32.dataCharToWord (toLower (Bech32.dataCharFromWord w))
                     === Just w
 
-    describe "Roundtrip (charToWord5 . word5ToChar)" $ do
-        it "can perform roundtrip character set conversion (upper-case)" $
-            property $ \w ->
-                Bech32.charToWord5 (toUpper (Bech32.word5ToChar Arr.! w))
-                    === Just w
+    describe "Constructors produce valid values" $ do
+
+        it "dataPartFromBytes" $
+            property $ \bytes ->
+                dataPartIsValid (dataPartFromBytes bytes) === True
```

Comment:
Note that there's little value in asserting `=== True`, the only counter example this can print is `False`,

Reply:
> Note that there's little value in asserting `=== True`, the only counter example this can print is `False`.

Understood. I've added another commit which improves the display of counterexamples (in all three constructor tests).